pkg/bundle/util.go

```diff
@@ -44,52 +44,72 @@ func bundleHasCondition(bundle *trustapi.Bundle, condition trustapi.BundleCondit
 
 // setBundleCondition updates the bundle with the given condition.
 // Will overwrite any existing condition of the same type.
-// ObservedGeneration of the condition will be set to the Generation of the
-// bundle object.
 // LastTransitionTime will not be updated if an existing condition of the same
 // Type and Status already exists.
-func (b *bundle) setBundleCondition(bundle *trustapi.Bundle, condition trustapi.BundleCondition) {
-	condition.LastTransitionTime = &metav1.Time{Time: b.clock.Now()}
-	condition.ObservedGeneration = bundle.Generation
-
-	var updatedConditions []trustapi.BundleCondition
-	for _, existingCondition := range bundle.Status.Conditions {
-		// Ignore any existing conditions which don't match the incoming type and
-		// add back to set.
-		if existingCondition.Type != condition.Type {
-			updatedConditions = append(updatedConditions, existingCondition)
+func (b *bundle) setBundleCondition(
+	existingConditions []trustapi.BundleCondition,
```

Comment:
Do we gain something by supplying the same slice two times (pointer and non-pointer), or can this parameter be removed? Please let me know if you think we've had enough discussion on this. 😉

Reply:
This is because in the SSA PR, these arrays will be different.
The first array will be the conditions that we get from the API server, the second array are the conditions in the patch object.

Reply:
Aha, thanks for the explanation!